custom_components/waste_collection_schedule/waste_collection_schedule/source/bendigo_vic_gov_au.py

```diff
@@ -0,0 +1,253 @@
+from waste_collection_schedule.exceptions import SourceArgumentNotFound
+
+TITLE = "City of Greater Bendigo"
+DESCRIPTION = "Source for City of Greater Bendigo rubbish collection."
+URL = "https://www.bendigo.vic.gov.au"
+
+# Geographic boundaries for Bendigo area
+BENDIGO_BOUNDS = {
+    "min_lat": -37.07,
+    "min_lon": 144.03,
+    "max_lat": -36.39,
+    "max_lon": 144.86
+}
+
+# API endpoints
+ZONES_API_URL = "https://d2nozjvesbm579.cloudfront.net/ogr2ogr?source=data.gov.au/bendigo/cogb-garbage_collection_zones.shz"
+
+# Test cases for validation
+TEST_CASES = {
+    "Bunnings Epsom": { 
+        "latitude": -36.701755710607394,
+        "longitude": 144.31310883736967
+    },
+    "Bunnings Bendigo": {
+        "latitude": -36.808262837180514,
+        "longitude": 144.24269331664098
+    },
+    "Alfa Kitchen Bendigo": {
+        "latitude": -36.758540554036315, 
+        "longitude": 144.2818129235716
+    },
+    # "Boundary Test - Friday Calendar B": {
+    #     # Point exactly halfway between [144.2876297, -36.7616511] and [144.2855719, -36.7598316]
+    #     "latitude": -36.76074135,  # (-36.7616511 + -36.7598316) / 2
+    #     "longitude": 144.2866008   # (144.2876297 + 144.2855719) / 2
+    # }
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+WASTE_NAMES = {
+    "waste": "General Waste",
+    "recycle": "Recycling",
+    "green": "Green Waste"
+}
+
+ICON_MAP = {
+    "waste": "mdi:trash-can",
+    "recycle": "mdi:recycle",
+    "green": "mdi:leaf"
+}
+
+
+class Source:
+    def __init__(self, latitude: float, longitude: float):
+        """Initialize source with latitude and longitude coordinates.
+        
+        Args:
+            latitude: Latitude coordinate (-37.07 to -36.39)
+            longitude: Longitude coordinate (144.03 to 144.86)
+        """
+        try:
+            # Convert inputs to float if they're strings
+            self._latitude = float(latitude)
+            self._longitude = float(longitude)
+            
+            if not BENDIGO_BOUNDS["min_lat"] <= self._latitude <= BENDIGO_BOUNDS["max_lat"]:
+                raise SourceArgumentNotFound(
+                    "latitude",
+                    str(self._latitude),
+                    f"Latitude must be between {BENDIGO_BOUNDS['min_lat']} and {BENDIGO_BOUNDS['max_lat']}"
+                )
+            if not BENDIGO_BOUNDS["min_lon"] <= self._longitude <= BENDIGO_BOUNDS["max_lon"]:
+                raise SourceArgumentNotFound(
+                    "longitude",
+                    str(self._longitude),
+                    f"Longitude must be between {BENDIGO_BOUNDS['min_lon']} and {BENDIGO_BOUNDS['max_lon']}"
+                )
+        except (ValueError, TypeError) as e:
+            raise SourceArgumentNotFound(
+                "coordinates",
+                f"({latitude}, {longitude})",
+                f"Invalid coordinate format. Please provide numeric values. Error: {str(e)}"
+            )
```

Comment:
This does not work like this just raise a normal exception here

As Ha config will try to match the exception to an input filed to show the error message next to this, but there is no field coordinates 

```suggestion
            raise Exception(
                f"Invalid coordinate format. Please provide numeric values. Error: {str(e)}"
            )
```